terraform/modules/front_door/access_log_bucket.tf

```diff
@@ -0,0 +1,91 @@
+#tfsec:ignore:aws-s3-enable-bucket-encryption: access log buckets not compatible with kms encryption, see https://docs.aws.amazon.com/AmazonS3/latest/userguide/troubleshooting-server-access-logging.html
+#tfsec:ignore:aws-s3-enable-bucket-logging: access log bucket doesn't need access logging itself
+#tfsec:ignore:aws-s3-encryption-customer-key: access log buckets not compatible with kms encryption, see https://docs.aws.amazon.com/AmazonS3/latest/userguide/troubleshooting-server-access-logging.html
+resource "aws_s3_bucket" "load_balancer_access_logs" {
+  #checkov:skip=CKV2_AWS_62: no need for event notifications
+  #checkov:skip=CKV_AWS_18: access log bucket doesn't need access logging itself
+  #checkov:skip=CKV_AWS_145: access log buckets not compatible with kms encryption, see https://docs.aws.amazon.com/AmazonS3/latest/userguide/troubleshooting-server-access-logging.html
+  #checkov:skip=CKV_AWS_144: cross region replication not required for access logs
+  bucket = "${var.prefix}-load-balancer-logs"
+}
+
+resource "aws_s3_bucket_public_access_block" "load_balancer_access_logs" {
+  bucket = aws_s3_bucket.load_balancer_access_logs.id
+
+  block_public_acls       = true
+  block_public_policy     = true
+  ignore_public_acls      = true
+  restrict_public_buckets = true
+}
+
+data "aws_elb_service_account" "current" {}
+
+resource "aws_s3_bucket_policy" "force_ssl_and_allow_access_logs" {
+  bucket = aws_s3_bucket.load_balancer_access_logs.id
+
+  policy = jsonencode({
+    Version = "2012-10-17",
+    Statement = [
+      {
+        Sid       = "AllowSSLRequestsOnly",
+        Action    = "s3:*",
+        Effect    = "Deny",
+        Principal = "*",
+        Resource = [
+          aws_s3_bucket.load_balancer_access_logs.arn,
+          "${aws_s3_bucket.load_balancer_access_logs.arn}/*"
+        ],
+        Condition = {
+          Bool = {
+            "aws:SecureTransport" = "false"
+          }
+        },
+        Sid    = "AllowLoadBalancerAccessLogs",
+        Action = "s3:PutObject",
+        Effect = "Allow",
+        Principal = {
+          AWS = data.aws_elb_service_account.current.arn

```

Comment:
Here AWS says you need to put the **account ID of the AWS region** your load balancer is in (not the account ID you've made the load balancer in itself), see "Attach a policy to your S3 bucket - Regions available before Aug 2022" section [here](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/enable-access-logging.html)

The terraform above is a programmatic way of getting this information rather than hardcoding it 😀